### PR TITLE
docs for build dependencies and managed paths

### LINKED ISSUE #3250: Document webpack change: add build dependencies
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/9567

---

**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
* `cache.managedPaths` is an array of package-manager only managed paths. webpack will avoid hashing/timestamping them, assume the version is unique and will use this as snapshot. (for both memory and filesystem cache) Defaults to the `node_modules` webpack is inside of.
* `cache.buildDependencies` is an object of arrays of additional code dependencies of the build. webpack will use an hash of these items and all dependencies to invalidate the filesystem cache. `cache.buildDependencies.defaultWebpack` defaults to `webpack/lib` to get all dependencies of webpack.
* It's recommended to set `cache.buildDependencies.config: [__filename]` in `webpack.config.js` to get the config and all dependencies.

### PR DESCRIPTION
**Pull request by niravasher**
Closes #3250 

[1]: https://cla.js.foundation/webpack/webpack.js.org
[2]: https://webpack.js.org/contribute/writers-guide/

### CODE REVIEW

File: src/content/configuration/other-options.md [hunk 1]
```diff
@@ -125,6 +126,34 @@ module.exports = {
 
 W> The final location of the cache is a combination of `cache.cacheDirectory` + `cache.name`.
 
+### `cache.buildDependencies`
+
```
Comment (jeffin143) [inline src/content/configuration/other-options.md]: duplicate of #3398

**Comment by niravasher** [comment]
#3398 has merge conflicts.

**Comment by niravasher** [comment]
@jamesgeorge007 @montogeek @EugeneHlushko what else is required to merge the PR? Please suggest required changes

File: src/content/configuration/other-options.md [hunk 2]
```diff
@@ -125,6 +126,34 @@ module.exports = {
...
+      config: [__filename]
+      // By default webpack and loaders are build dependencies
+    }
+  }
+};
+```
+
+### `cache.managedPaths`
+
+`[path] = ['./node_modules']`
```
Comment (EugeneHlushko) [inline src/content/configuration/other-options.md:153]:
```suggestion
`[string] = ['./node_modules']`
```

File: src/content/configuration/other-options.md [hunk 3]
```diff
@@ -146,6 +147,34 @@ module.exports = {
 };
 ```
 
+### `cache.buildDependencies`
+
+`object`
+
+`cache.buildDependencies` is an object of arrays of additional code dependencies of the build. webpack will use a hash of each of these items and all dependencies to invalidate the filesystem cache.
```
Comment (EugeneHlushko) [inline src/content/configuration/other-options.md:154]:
```suggestion
`cache.buildDependencies` is an object of arrays of additional code dependencies for the build. webpack will use a hash of each of these items and all dependencies to invalidate the filesystem cache.
```

**Comment by EugeneHlushko** [comment]
Thanks!

**Pull request merged**